src/Nexus.ts

```diff
@@ -155,6 +178,10 @@ class Nexus {
   private mQuota: Quota;
   private mValidationResult: types.IValidateKeyResponse;
   private mRateLimit: { daily: number, hourly: number } = { daily: 1000, hourly: 100 };
+  private oAuthCredentials: types.IOAuthCredentials;
+  private oAuthConfig: types.IOAuthConfig; 
+  private onNewOAuthCredentialsHandler: types.OnNewOauthCredentialsHandler;
```

Comment:
This doesn't _have_ to change but usually when I have handler callbacks I have an array of them. In practice it's unlikely that the client will want to attach multiple handlers to this event but who knows?